app/models/manager_refresh/inventory_collection.rb

```diff
@@ -378,6 +378,10 @@ def data_collection_finalized?
       data_collection_finalized
     end
 
+    def supports_sti?
+      @supports_sti_cache ||= model_class.column_names.include?("type")
```

Comment:
@Ladas this is the true/false/nil pattern, so you probably need something like the following so you are memoizing and not recalculating when value is `false`
```ruby
def supports_sti?
  if @supports_sti_cache.nil?
    @supports_sti_cache = model_class.column_names.include?("type")
  end
  @supports_sti_cache
end
```